packages/gatsby/src/utils/eslint-rules/limited-exports-page-templates.ts

```diff
@@ -0,0 +1,80 @@
+import { Rule } from "eslint"
+import { Node, Identifier } from "estree"
+import { store } from "../../redux"
+import { isPageTemplate } from "../eslint-rules-helpers"
+
+function isTemplateQuery(node: Node, varName: string | undefined): boolean {
+  // Checks for:
+  // const query = graphql``
+  // export { query }
+  if (
+    node?.type === `ExportNamedDeclaration` &&
+    node?.declaration === null &&
+    varName
+  ) {
+    // For export { foobar } the declaration will be null and specifiers exists
+    const nonQueryExports = node.specifiers.find(
+      e => e.exported.name !== varName
+    )
+    return !nonQueryExports
```

Comment:
`.some`?

```suggestion
    return node.specifiers.some(
      e => e.exported.name === varName
    )
```

I don't quite get this one, actually. Right now, it searches for an occurrence of an export that is not var name. If it finds it, it returns false ("this is not a template query"). But I think it should return true if var name was part of the specifiers? Regardless of what other exports were in there?

Reply:
We only want `export { query }` at the end to be allowed. So if one has `export { stuff, query }` it's invalid as only the named export of your graphql call is allowed.

So my logic behind this was:

1) search exports that are not graphql query
2) If it finds something, this is truthy
3) return false so that context.report is triggered

So I think the problem here is that it probably shouldn't be part of `isTemplateQuery` and rather it's own function as it checks if there are more named exports than allowed.